tests/web/rpc/v1/test_endpoint_trace_item_attribute_names.py

```diff
@@ -0,0 +1,342 @@
+    EndpointTraceItemAttributeNames,
+)
+from tests.base import BaseApiTest
+from tests.helpers import write_raw_unprocessed_events
+
+BASE_TIME = datetime.now(UTC).replace(minute=0, second=0, microsecond=0) - timedelta(
+    hours=3
+)
+
+
+def populate_eap_spans_storage(num_rows: int) -> None:
+    """
+    Fills the eap_spans storage with data for num_rows spans.
+    Each span will have at least 10 unique tags.
+    """
+
+    def generate_span_event_message(id: int) -> Mapping[str, Any]:
+        res = {
+            "description": "/api/0/relays/projectconfigs/",
+            "duration_ms": 152,
+            "event_id": "d826225de75d42d6b2f01b957d51f18f",
+            "exclusive_time_ms": 0.228,
+            "is_segment": True,
+            "data": {},
+            "measurements": {},
+            "organization_id": 1,
+            "origin": "auto.http.django",
+            "project_id": 1,
+            "received": 1721319572.877828,
+            "retention_days": 90,
+            "segment_id": "8873a98879faf06d",
+            "sentry_tags": {
+                "category": "http",
+            },
+            "span_id": uuid.uuid4().hex,
+            "tags": {
+                "http.status_code": "200",
+            },
+            "trace_id": uuid.uuid4().hex,
+            "start_timestamp_ms": int(BASE_TIME.timestamp() * 1000),
+            "start_timestamp_precise": BASE_TIME.timestamp(),
+            "end_timestamp_precise": BASE_TIME.timestamp() + 1,
+        }
+        for i in range(id * 10, id * 10 + 10):
+            res["tags"][f"a_tag_{i:03}"] = "blah"  # type: ignore
+            res["measurements"][f"b_measurement_{i:03}"] = {"value": 10}  # type: ignore
+        return res
+
+    spans_storage = get_storage(StorageKey("eap_spans"))
+    messages = [generate_span_event_message(i) for i in range(num_rows)]
+    write_raw_unprocessed_events(spans_storage, messages)  # type: ignore
+
+
+@pytest.fixture(autouse=True)
+def setup_teardown(clickhouse_db: None, redis_db: None) -> None:
+    populate_eap_spans_storage(num_rows=3)
+
+
+@pytest.mark.clickhouse_db
+@pytest.mark.redis_db
+class TestTraceItemAttributeNames(BaseApiTest):
+    def test_basic(self) -> None:
+        req = TraceItemAttributeNamesRequest(
+            meta=RequestMeta(
+                project_ids=[1, 2, 3],
+                organization_id=1,
+                cogs_category="something",
+                referrer="something",
+                start_timestamp=Timestamp(
+                    seconds=int((BASE_TIME - timedelta(days=1)).timestamp())
+                ),
+                end_timestamp=Timestamp(
+                    seconds=int((BASE_TIME + timedelta(days=1)).timestamp())
+                ),
+            ),
+            limit=1000,
+            type=AttributeKey.Type.TYPE_STRING,
+            value_substring_match="",
+        )
+        res = EndpointTraceItemAttributeNames().execute(req)
+
+        expected = []
+        for i in range(30):
```

Comment:
where is this magic number coming from?

Reply:
I introduced some constants like `TOTAL_GENERATED_SPANS` to get rid of the magic numbers